src/evaluator/datatypes.ts

```diff
@@ -9,25 +8,26 @@ export enum EvalDataType {
 
 export interface Closure {
   variant: EvalDataType.Closure;
-  env: Environment | undefined;
+  env: Environment;
```

Comment:
Can I clarify why we've taken out undefined here?

Reply:
Environment is now the union of NonemptyEnvironment and undefined